src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs

```diff
@@ -0,0 +1,14 @@
+﻿using FluentValidation;
+
+namespace AeroSharp.DataAccess.Validation;
+
+internal sealed class MapConfigurationValidator : AbstractValidator<MapConfiguration>
+{
+    public MapConfigurationValidator()
+    {
+        RuleFor(mapConfiguration => mapConfiguration.CreateOnly)
+            .Equal(false)
+            .When(mapConfiguration => mapConfiguration.UpdateOnly)
+            .WithMessage("CreateOnly and UpdateOnly cannot both be true.");
```

Comment:
Don't we also need to add a rule for `mapConfiguration.UpdateOnly`? I think this only guarantees that `CreateOnly` is false when `UpdateOnly` is true, but not the other way around.

Reply:
We don't explicitly need to include that. This validation only runs when `CreateOnly` is `true` and ensures that `UpdateOnly` is set to `false. When `CreateOnly` is set to `false`, it doesn't matter what the value of `UpdateOnly` is. 

The associated unit tests confirm this, but I'd be happy to add the other scenario for the sake of self-documenting code and clarity. What do you think?

Reply:
Ah, this makes sense. I don't think it needs to be explicit anywhere. Sounds good to me!